packages/vue-query/src/useInfiniteQuery.ts

```diff
@@ -72,13 +76,21 @@ export function useInfiniteQuery<
   TError = RegisteredError,
   TData = TQueryFnData,
   TQueryKey extends QueryKey = QueryKey,
+  TPageParam = unknown,
 >(
-  options: UseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
+  options: UseInfiniteQueryOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryFnData,
+    TQueryKey,
+    TPageParam
+  >,
   queryClient?: QueryClient,
 ): UseInfiniteQueryReturnType<TData, TError> {
   const result = useBaseQuery(
     InfiniteQueryObserver as typeof QueryObserver,
-    options,
+    options as any,
```

Comment:
@DamianOsipiuk I had to stick an `as any` here but I don't really understand why. Note that the previous types had an issue I think:

`UseInfiniteQueryOptions<TQueryFnData, TError, TData, TQueryKey>` wasn't really right as it was missing a generic, so it was passing `TQueryKey` in the "wrong spot". I only noticed wen adding `TPageParam`. Could you have a look please?

Reply:
@TkDodo Fixed

Reply:
🙏